common/web/types/src/util/util.ts

```diff
@@ -15,3 +15,13 @@ export function boxXmlArray(o: any, x: string): void {
     }
   }
 }
+
+const MATCH_HEX_ESCAPE = /\\u{([0-9a-fA-F]{1,5})}/g;
+
+export function unescapeString(s: string): string {
+  if(!s) return s;
+
+  s = s.replaceAll(MATCH_HEX_ESCAPE, (str,hex) => String.fromCodePoint(Number.parseInt(hex, 16)));
```

Comment:
This will throw `RangeError` on invalid values such as `\u{110000}`. How do we want to handle that?

Reply:
Perhaps it should be caught here and thrown as its own `CompilerError` specifically stating an invalid escape.

Reply:
That would be good. Then the compiler will handle `CompilerError`s and reframe them through the compiler logging. A real `RangeError` elsewhere will then be unhandled, as it should be.

We also want to block surrogate pairs and illegal chars somewhere, but not here.

Reply:
it's thrown as an `UnescapeError` now.